src/query/executor/request.go

```diff
@@ -108,63 +103,29 @@ func (r *Request) plan(ctx context.Context, nodes parser.Nodes, edges parser.Edg
 
 	pp, err := plan.NewPhysicalPlan(lp, r.engine.store, r.params)
 	if err != nil {
-		sp.finish(err)
 		return plan.PhysicalPlan{}, err
 	}
 
 	if r.params.Debug {
 		logging.WithContext(ctx).Info("physical plan", zap.String("plan", pp.String()))
 	}
 
-	sp.finish(nil)
 	return pp, nil
 }
 
-func (r *Request) execute(ctx context.Context, pp plan.PhysicalPlan) (*ExecutionState, error) {
-	sp := startSpan(r.engine.metrics.executingHist, r.engine.metrics.executing)
+func (r *Request) generateExecutionState(ctx context.Context, pp plan.PhysicalPlan) (*ExecutionState, error) {
+	sp, ctx := opentracing.StartSpanFromContext(ctx, "generateExecutionState")
```

Comment:
Not sure about the naming semantics for opentracing, but do they recommend camel case or using `_` or something else?